### ISSUE TITLE
web.client needs documentation and/or data-checking for method argument

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10446]
|<img alt="mcfletch's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mcfletch reported|
|-|-|
|Trac ID|trac#687|
|Type|defect|
|Created|2004-08-25 01:34:48Z|

<details><summary>Searchable metadata</summary>

```
trac-id__687 687
type__defect defect
reporter__mcfletch mcfletch
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1093397688000000 1093397688000000
changetime__1212969002000000 1212969002000000
version__ 
owner__ 
cc__jknight cc__mcfletch cc__oubiwann
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="mcfletch's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|mcfletch commented|
|-|-|

```
#!html
<pre>
Have just tracked down a problem with a small application I'm writing.  In
short, the "method" argument to twisted.web.client.HTTPDownloader.__init__ is
apparently getting passed straight through to the HTTP request without any
checking.  I was passing in "get" and "post", and many servers (not all) were
returning 501 errors.  (Following some sample code I found somewhere IIRC).

Since there may be (rare) use-cases where one wants to pass in an unknown
method, I don't suppose that forcing the parameter to be from a given set is
acceptable, but at a minimum the __init__ method should have documentation to
the effect that the method should be "POST" or "GET" for normal usage.

Also, the web.client.getPage function really should have a note regarding the
fact that it supports https:// protocols without any extra work.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Assigning tentatively to jknight
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Assigning tentatively to jknight
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Docs, eh

```
Index: twisted/web/client.py
===================================================================
--- twisted/web/client.py       (revision 23898)
+++ twisted/web/client.py       (working copy)
@@ -190,6 +190,12 @@
     @type response_headers: dict
     @ivar response_headers: The headers that were specified in the
           response from the server.
+
+    @type method: str
+    @ivar method: The HTTP method to use in the request.  This should be one of
+        OPTIONS, GET, HEAD, POST, PUT, DELETE, TRACE, or CONNECT (case
+        matters).  Other values may be specified if the server being contacted
+        supports them.
     """

     protocol = HTTPPageGetter
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @exarkun|
|-|-|

Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [23914]) Apply patch adding `method` to the `HTTPClientFactory` class docstring

Author: exarkun
Reviewer: therve
Fixes: [#10446](https://github.com/twisted/twisted/issues/10446)

Document the expected valid values for the `method` attribute of `HTTPClientFactory`
and note that case matters.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
